server/src/main/java/org/opensearch/search/SearchHit.java

```diff
@@ -560,14 +544,38 @@ public String getClusterAlias() {
     }
 
     public void matchedQueries(String[] matchedQueries) {
-        this.matchedQueries = matchedQueries;
+        if (matchedQueries != null) {
+            for (String query : matchedQueries) {
+                this.matchedQueries.put(query, Float.NaN);
+            }
+        }
+    }
+
+    public void matchedQueriesWithScores(Map<String, Float> matchedQueries) {
+        if (matchedQueries != null) {
+            this.matchedQueries = matchedQueries;
+        }
     }
 
     /**
      * The set of query and filter names the query matched with. Mainly makes sense for compound filters and queries.
      */
     public String[] getMatchedQueries() {
-        return this.matchedQueries;
+        return matchedQueries == null ? new String[0] : matchedQueries.keySet().toArray(new String[0]);
+    }
+
+    /**
+     * @return The score of the provided named query if it matches, {@code null} otherwise.
```

Comment:
Could you please include a comment here (javadocs) that if `include_named_queries_score` is not set, the `Float.NaN` is going to be returned for each named query